cleanlab/filter.py

```diff
@@ -0,0 +1,687 @@
+# Copyright (C) 2017-2022  Cleanlab Inc.
+# This file is part of cleanlab.
+# 
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# 
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+# 
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+
+## Filter (previously in Cleanlab 1.0, this module was called Pruning)
+
+# #### Contains methods for estimating the latent indices of all label issues.
+# This code uses advanced multiprocessing to speed up computation.
+# see: https://research.wmz.ninja/articles/2018/03/ (link continued below)
+# on-sharing-large-arrays-when-using-pythons-multiprocessing.html
+
+
+
+
+from sklearn.metrics import confusion_matrix
+from sklearn.preprocessing import MultiLabelBinarizer
+import multiprocessing
+from multiprocessing.sharedctypes import RawArray
+import sys
+
+from cleanlab.rank import order_label_issues
+from cleanlab.util import (value_counts, round_preserving_row_totals,
+                           onehot2int, int2onehot, )
+import numpy as np
+import warnings
+
+# tqdm is a module used to print time-to-complete when multiprocessing is used.
+# This module is not necessary, and therefore is not a package dependency, but 
+# when installed it improves user experience for large datasets.
+from cleanlab.count import calibrate_confident_joint
+
+try:
+    import tqdm
+
+    tqdm_exists = True
+except ImportError as e:
+    tqdm_exists = False
+
+    w = '''To see estimated completion times for methods in cleanlab.filter, "pip install tqdm".'''
+    warnings.warn(w)
+
+# Leave at least 1 example in each class after filtering/pruning, even if noise estimates are larger
+MIN_NUM_PER_CLASS = 1
+
+# For python 2/3 compatibility, define pool context manager
+# to support the 'with' statement in Python 2
+if sys.version_info[0] == 2:
```

Comment:
Don't need this block anymore; can get rid of the whole if/else and replace `multiprocessing_context` with `multiprocessing.Pool`